core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java

```diff
@@ -129,6 +129,11 @@ public void close() throws IOException {
     targetStream.close();
   }
 
+  @Override
+  public long storedLength() throws IOException {
+    return targetStream.storedLength();
```

Comment:
this works. But to be on the safe side, might have to be replaced with `targetStream.getPos()`. If the target stream is not length preserving (for some strange reason), then the AES GCM Stream decryption will be broken by this value.

Reply:
This looks right to me, if the underlying stream is not length preserving it should also have a custom storedLength impl

Reply:
Ok, lets leave it like this. When (or rather if) we'll have this situation, we'll handle the required scaffolding.

Reply:
I agree. This looks good to me.